pkg/controller/machine/controller.go

```diff
@@ -17,75 +17,96 @@ limitations under the License.
+	}
 
-	// nodeName is the name of the node on which the machine controller is running, if not present, it is loaded from NODE_NAME.
-	nodeName string
+	return r
 }
 
-// Init initializes the controller and is called by the generated code
-// Register watches for additional resource types here.
-func (c *MachineControllerImpl) Init(arguments sharedinformers.ControllerInitArguments, actuator Actuator) {
-	// Use the lister for indexing machines labels
-	c.lister = arguments.GetSharedInformers().Factory.Cluster().V1alpha1().Machines().Lister()
-
-	if c.nodeName == "" {
-		c.nodeName = os.Getenv(NodeNameEnvVar)
-		if c.nodeName == "" {
-			glog.Warningf("environment variable %v is not set, this controller will not protect against deleting its own machine", NodeNameEnvVar)
-		}
+// add adds a new Controller to mgr with r as the reconcile.Reconciler
+func add(mgr manager.Manager, r reconcile.Reconciler) error {
+	// Create a new controller
+	c, err := controller.New("machine-controller", mgr, controller.Options{Reconciler: r})
+	if err != nil {
+		return err
 	}
-	clientset, err := clientset.NewForConfig(arguments.GetRestConfig())
+
+	// Watch for changes to Machine
+	err = c.Watch(&source.Kind{Type: &clusterv1.Machine{}}, &handler.EnqueueRequestForObject{})
 	if err != nil {
-		glog.Fatalf("error creating machine client: %v", err)
+		return err
 	}
-	c.clientSet = clientset
-	c.kubernetesClientSet = arguments.GetSharedInformers().KubernetesClientSet
 
-	c.linkedNodes = make(map[string]bool)
-	c.cachedReadiness = make(map[string]bool)
+	return nil
+}
+
+var _ reconcile.Reconciler = &ReconcileMachine{}
 
-	c.actuator = actuator
+// ReconcileMachine reconciles a Machine object
+type ReconcileMachine struct {
+	client.Client
+	scheme *runtime.Scheme
 
-	// Start watching for Node resource. It will effectively create a new worker queue, and
-	// reconcileNode() will be invoked in a loop to handle the reconciling.
-	ni := arguments.GetSharedInformers().KubernetesFactory.Core().V1().Nodes()
-	arguments.GetSharedInformers().Watch("NodeWatcher", ni.Informer(), nil, c.reconcileNode)
+	actuator Actuator
+
+	// nodeName is the name of the node on which the machine controller is running, if not present, it is loaded from NODE_NAME.
+	nodeName string
 }
 
-// Reconcile handles enqueued messages. The delete will be handled by finalizer.
-func (c *MachineControllerImpl) Reconcile(machine *clusterv1.Machine) error {
-	// Deep-copy otherwise we are mutating our cache.
-	m := machine.DeepCopy()
+// Reconcile reads that state of the cluster for a Machine object and makes changes based on the state read
+// and what is in the Machine.Spec
+// +kubebuilder:rbac:groups=cluster.k8s.io,resources=machines,verbs=get;list;watch;create;update;patch;delete
+func (r *ReconcileMachine) Reconcile(request reconcile.Request) (reconcile.Result, error) {
```

Comment:
Just wondering about the lifecycle of the controllers with this approach, am I correct that these are not generated, just using a pattern similar to what kubebuilder would generate for a fresh project, and then manually maintained going forward?